numpy/__init__.py

```diff
@@ -182,9 +185,35 @@
     oldnumeric = 'removed'
     numarray = 'removed'
 
-    # We don't actually use this ourselves anymore, but I'm not 100% sure that
-    # no-one else in the world is using it (though I hope not)
-    from .testing import Tester
+    if sys.version_info[:2] >= (3, 7):
+        # Importing Tester requires importing all of UnitTest which is not a
+        # cheap import Since it is mainly used in test suits, we lazy import it
+        # here to save on the order of 10 ms of import time for most users
+        #
+        # The previous way Tester was imported also had a side effect of adding
+        # the full `numpy.testing` namespace
+        #
+        # module level getattr is only supported in 3.7 onwards
+        # https://www.python.org/dev/peps/pep-0562/
+        def __getattr__(attr):
+            if attr == 'testing':
+                import numpy.testing as testing
+                return testing
+            elif attr == 'Tester':
+                from .testing import Tester
+                return Tester
+            else:
+                raise AttributeError(
+                    "module %s has no attribute $s".format(__name__, attr))
+
+
+        def __dir__():
+            return __all__ + ['Tester']
```

Comment:
This should probably also add `testing`? If there is a test for that somewhere for `Tester`, can you add this too?

Reply:
```
In [3]: 'testing' in dir(numpy)                                                                          
Out[3]: True
In [4]: numpy.__version__                                                       
Out[4]: '1.17.0'
```

you are right.

Reply:
```
AttributeError: module 'numpy.testing' has no attribute '__module__'
```

Reply:
I think I fixed up the tests. Hopefully that should address this concern.